daemon/api_find_test.go

```diff
@@ -423,6 +427,45 @@ func (s *findSuite) TestFindBadQueryReturnsCorrectErrorKind(c *check.C) {
 	c.Check(rsp.Result.(*daemon.ErrorResult).Kind, check.Equals, client.ErrorKindBadQuery)
 }
 
+func (s *findSuite) TestFindNetworkErrorsReturnCorrectErrorKind(c *check.C) {
+	s.daemon(c)
+
+	req, err := http.NewRequest("GET", "/v2/find?q=query", nil)
+	c.Assert(err, check.IsNil)
```

Comment:
I've often used httptest.NewRequest() here, which creates a request closer to what a server-side Request would look like. It also avoids the need to check for errors.

Reply:
we don't seem in the habit to use this much, I see your new code did. Maybe we should refactor everything to use it more, it seems to be meant exactly for this as you said, but it seems its own thing.

Reply:
I agree that it would be worth cleaning up the other call sites in a separate PR.  It just seemed like it might be worth using `httptest.NewRequest` in new code.

Of course, if this gets merged quickly, then ultimately it won't make much of a difference.